kapitan/cli.py

```diff
@@ -430,6 +439,32 @@ def ref_write(args, ref_controller):
         secret_obj = GoogleKMSSecret(data, key, encode_base64=args.base64)
         tag = '?{{gkms:{}}}'.format(token_path)
         ref_controller[tag] = secret_obj
+#
+# Azure KMS marker -- remove it
+#
+    elif token_name.startswith("azkms:"):
+        type_name, token_path = token_name.split(":")
+        key = args.key
+        vault = args.vault
+        if args.target_name:
+            inv = inventory_reclass(args.inventory_path)
+            kap_inv_params = inv['nodes'][args.target_name]['parameters']['kapitan']
+            if 'secrets' not in kap_inv_params:
+                raise KapitanError("parameters.kapitan.secrets not defined in {}".format(args.target_name))
+
+            key = kap_inv_params['secrets']['azkms']['key']
+            vault = kap_inv_params['secrets']['azkms']['vault']
+        if not key:
+            raise KapitanError("No KMS key specified. Use --key or specify it in parameters.kapitan.secrets.azkms.key and use --target-name")
+        if not vault:
+            raise KapitanError("No KMS vault specified. Use --vault or specify it in parameters.kapitan.secrets.azkms.vault and use --target-name")
```

Comment:
rename `--vault` here too.

Reply:
ok

Reply:
can I do something like this ? for both key & vault 
  if os.environ.get('AZ_KEY_NAME') is set:
        then override inventory key